client/data/site-settings/use-site-settings-query.ts

```diff
@@ -0,0 +1,15 @@
+import { useQuery } from '@tanstack/react-query';
+import wp from 'calypso/lib/wp';
+
+const useSiteSettingsQuery = ( siteId: number | null ) =>
+	useQuery( {
+		queryKey: [ 'site-settings', siteId ],
+		queryFn: () => wp.req.get( `/sites/${ siteId }/settings`, { apiVersion: '1.4' } ),
```

Comment:
The existing `wrapSettingsForm` HoC has some complexity that we might not be considering here, like fetching Jetpack settings separately and combining them:

https://github.com/Automattic/wp-calypso/blob/270e72e2277b5ea3209108bf4b89544763fc7942/client/my-sites/site-settings/wrap-settings-form.jsx#L385